code/modules/vehicles/ridden.dm

```diff
@@ -71,7 +71,7 @@
 				message_cooldown = world.time + 5 SECONDS
 			return FALSE
 	if(arms_required)
-		if(user.usable_hands < arms_required)
+		if(user.usable_hands < arms_required || HAS_TRAIT(user, TRAIT_FLOORED) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || HAS_TRAIT(user, TRAIT_INCAPACITATED))
```

Comment:
While this change is good balance-wise, these traits were done precisely to add granularity to checks and give them meaning.
This check is about arms being required, and in this context `TRAIT_HANDS_BLOCKED` makes all sense, but `TRAIT_FLOORED` doesn't (this is related to legs / movement appendages) and `TRAIT_INCAPACITATED` is questionable.

Could you add independent checks for the others?
The best way for that kind of thing would be to move the boolean `arms_required` into a bitflag holder. something like `ride_check_flags` that can hold flags such as `REQUIRES_HANDS` and `REQUIRES_LEGS`.